kube/m3nsch.yaml

```diff
@@ -0,0 +1,130 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: m3nsch-server-config
+data:
+  m3nsch_server.yaml: |+
+    server:
+      listenAddress: 0.0.0.0:2580
+      debugAddress: 0.0.0.0:12580
+      cpuFactor: 0.9
+
+    metrics:
+      prometheus:
+        handlerPath: /metrics
+      samplingRate: 1.0
+      extended: detailed
+      sanitization: prometheus
+
+    m3nsch:
+      concurrency: 2000
+      numPointsPerDatum: 60
+
+    dbClient:
+      config:
+        service:
+          zone: default_zone
+          env: default_env
+          service: m3db
+          etcdClusters:
+            - zone: default_zone
+              endpoints:
+                - etcd-0.etcd:2379
+                - etcd-1.etcd:2379
+                - etcd-2.etcd:2379
+      writeConsistencyLevel: majority
+      readConsistencyLevel: unstrict_majority
+      writeTimeout: 10s
+      fetchTimeout: 15s
+      connectTimeout: 20s
+      writeRetry:
+          initialBackoff: 500ms
+          backoffFactor: 3
+          maxRetries: 2
+          jitter: true
+      fetchRetry:
+          initialBackoff: 500ms
+          backoffFactor: 2
+          maxRetries: 3
+          jitter: true
+      backgroundHealthCheckFailLimit: 4
+      backgroundHealthCheckFailThrottleFactor: 0.5
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: m3nsch-agent
+spec:
+  ports:
+  - name: agent
+    port: 2580
+  - name: debug
+    port: 12580
+  selector:
+    app: m3nsch
+    component: agent
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: m3nsch-server
+spec:
+  replicas: 1
```

Comment:
just a FYI, you can tune this - i.e. one client can kick off a job with `n` servers if you need more load.

Reply:
Yeah good point. I might change this to a `StatefulSet` so that we can get stable hostnames per-pod, since currently even if we kick off > 1 agents we'd have to look up each pod name since the service does random LB